### PR TITLE
Assign field ID to input catalog based on RA,DEC

### PR DESCRIPTION
**Pull request by dr-guangtou**
Add a standard alone script: `ssp_data.py` to prepare input lensing catalog.  

Right now, we only need to deal with HSC SSP data from `s16a_wide` datasets.  So it is easy to define a list of fields that are described by (RA, Dec) boxes and assign objects from input catalog to them. 

The current field ID is based on Felipe's choice: 

```
field_name = ['GAMA09H', 'GAMA15H', 'WIDE12H', 'HECTOMAP', 'XMM', 'VVDS', 'AEGIS']
field_id = [1, 2, 3, 4, 5, 6, 7]
```

Example: 

```
import lib.ssp_data as ssp_data

test_sample = ssp_data.prep_catalog_s16a(os.path.join(test_dir, 'stacked_sample.fits'))
```

### CODE REVIEW

File: lib/ssp_data.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        Name of the field
+    """
+    def __init__(self, ra_min, ra_max, dec_min, dec_max, name):
+        self.ra_min = ra_min
+        self.ra_max = ra_max
+        self.dec_min = dec_min
+        self.dec_max = dec_max
+        self.name = name
+
+        assert ra_min < ra_max, "# Wrong RA range !"
```
Comment (Christopher-Bradshaw) [inline lib/ssp_data.py]: We (me included) are mixing double and single quotes (see line below).

OCD is going to require me to do something about it at some point :)

**Comment by Christopher-Bradshaw** [comment]
Should we integrate this into our pipeline? I'm always worried about code that isn't routinely executed - good chance that it becomes slightly out of date and then hard to reintegrate later. We could drop the field field and have this regen it whenever we run precompute?

Maybe not worth the effort though.

Either way, LGTM :+1:

File: lib/ssp_data.py (hunk 1 above)
Comment (dr-guangtou) [reply lib/ssp_data.py]: Understand, I seem to always use double-ones as default, but feel free to change that, I have not preference.

**Comment by dr-guangtou** [comment]
@Christopher-Bradshaw 

We need field simply because the HSC SSP areas are very isolated and there is a long distance between each of them...so separate the computation in each field makes sense.  

In the long run, these fields will change and be connected.  We certainly want to come up with a more flexible way to define fields.  

I can imagine we have a script to prepare all the inputs.  

1) Define fields based on the input lens catalog. 
2) Select useful regions from sources and random catalogs, assign fields in a consistent way. 
3) Reweight the random to have the same redshift distribution.  
4) Save the data in `.npy` format.  

However, for the next year or so, we are only going to use the `s16a` data release, hence the current approach is just the easiest one.

**Comment by dr-guangtou** [comment]
Pull request merged, and the `song` branch is deleted.....I should use better branch name next time...

**Pull request merged**